packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -225,6 +234,10 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         }
         // TODO t15106389: add helper utility for fetching more data
         this._pendingRefetch = null;
+        this._relayContext = {
+          environment: this.context.relay.environment,
+          variables: fragmentVariables
```

Comment:
I could be wrong, but is it because here we are potentially exposing local variables into root variables?
we have `root argument` and `local argument`, only `root argument` should go to context; but here, the `fragmentVariables` is a mixed of both; I think we should filter it